spec/system/admin/enterprises_spec.rb

```diff
@@ -532,5 +533,81 @@
         end
       end
     end
+
+    context "white label settings" do
+      context "when the feature is disabled" do
+        before do
+          Flipper.disable(:white_label)
+          visit edit_admin_enterprise_path(distributor1)
+        end
+
+        it "does not show the white label settings" do
+          expect(page).not_to have_link "White Label"
+        end
+      end
+
+      context "when the feature is enabled" do
+        before do
+          Flipper.enable(:white_label)
+          visit edit_admin_enterprise_path(distributor1)
+
+          within(".side_menu") do
+            click_link "White Label"
+          end
+        end
+
+        it "set the hide_ofn_navigation preference for the current shop" do
+          expect(page).not_to have_content "LOGO USED IN SHOPFRONT"
+          check "Hide OFN navigation"
+          click_button 'Update'
+          expect(flash_message).to eq('Enterprise "First Distributor" has been successfully updated!')
+          expect(distributor1.reload.preferred_hide_ofn_navigation).to be true
+
+          visit edit_admin_enterprise_path(distributor1)
+          within(".side_menu") do
+            click_link "White Label"
+          end
+
+          expect(page).to have_content "LOGO USED IN SHOPFRONT"
+          uncheck "Hide OFN navigation"
+          click_button 'Update'
+          expect(flash_message).to eq('Enterprise "First Distributor" has been successfully updated!')
+          expect(distributor1.reload.preferred_hide_ofn_navigation).to be false
+        end
+
+        context "when white label is active via `hide_ofn_navigation`" do
+          before do
+            distributor1.update_attribute(:preferred_hide_ofn_navigation, true)
+
+            visit edit_admin_enterprise_path(distributor1)
+            within(".side_menu") do
+              click_link "White Label"
+            end
+          end
+
+          it "can updload the white label logo for the current shop" do
+            attach_file "enterprise_white_label_logo", white_logo_path
+            click_button 'Update'
+            expect(flash_message).to eq('Enterprise "First Distributor" has been successfully updated!')
+            expect(distributor1.reload.white_label_logo_blob.filename).to eq("logo-white.png")
+          end
+
+          context "when enterprise has a white label logo" do
```

Comment:
Hey @jibees ,

I was able to attach a file with:
```suggestion
          context "when enterprise has a white label logo" do
                      let(:white_logo_file) { Rails.root.join("public/default_images/ofn-logo.png") }

            before do
              attach_file "enterprise[white_label_logo]", white_logo_file
            end
```
After this change, we can see the file is attached (by running with the headless mode off), but the `Remove` button does not seem to appear... Maybe some code change is still needed?

Reply:
Sorry, I should have been clearer ;) 
I can attach a file to a the form:
https://github.com/jibees/openfoodnetwork/blob/10559-allow-upload-of-custom-header-logo/spec/system/admin/enterprises_spec.rb#L589

but what I'm not able to do is to set to a distributor a white label logo. 
I've the same issue here: 
https://github.com/jibees/openfoodnetwork/blob/10559-allow-upload-of-custom-header-logo/spec/system/consumer/white_label_spec.rb#L221 

(that's why I've a `xit` few lines after). 

So my question is: how can I set a white label logo (defined has an attachment [here](https://github.com/jibees/openfoodnetwork/blob/10559-allow-upload-of-custom-header-logo/app/models/enterprise.rb#L91)) to an enterprise in specs? 

Thanks! 🙏

Reply:
Oh, I see now - sorry, I misunderstood.

I've looked at https://github.com/jibees/openfoodnetwork/blob/10559-allow-upload-of-custom-header-logo/spec/system/consumer/white_label_spec.rb#L221 now.

I think the syntax is fine, and that the logo can be set in this way. However, I've noticed that the dimensions of the `app/webpacker/images/logo-white.png` are not the same as `/default_images/ofn-logo.png`. This seems to make the image not visible:

![image](https://user-images.githubusercontent.com/49817236/226322420-25b33603-79c0-4da5-8e56-8d0b2732c757.png)

But this is not due to syntax. If we try the same syntax for a file with correct dimensions - for example `app/webpacker/images/logo-australia.png` then it appears:

![image](https://user-images.githubusercontent.com/49817236/226323000-fc8c2720-debd-466e-be7f-d2ce2099301f.png)

All this should actually not really matter, as we are only asserting on the path - sorry for the detour :sweat_smile: But I was not sure we were able to upload files this way.

Ok, so now, knowing that images uploaded (i.e., updated on the correct `distributor`) successfully, the assertion on the path fails:

`expect(page).to have_selector "img[src*='#{white_logo_file}']"`

It only worked if I replaced the line above with:

`expect(page).to have_selector "img[src*='/logo-white.png']"`

Is this helpful?

Reply:
Nice, I think it's perfect now ;)
Thanks for the help, I missed that because I cannot see the image when running in non-headless mode, but it's because it's a white logo on a white nav bar! So _invisible_... Noob I was. But anyway, syntax was almost correct ;)

🙏